src/chainrules.jl

```diff
@@ -19,9 +19,15 @@ end
 function _Ω_∂_xlogy(x::Real, y::Real)
     logy = log(y)
     z = x * logy
-    Ω = iszero(x) && !isnan(y) ? zero(z) : z
-    ∂x = logy
-    ∂y = x / y
+    w = x / y
+    if iszero(x) && !isnan(y)
+        Ω = zero(z)
+        ∂y = zero(w)
+    else
+        Ω = z
+        ∂y = iszero(y) ? oftype(w, NaN) : w
+    end
+    ∂x = iszero(y) ? oftype(logy, NaN) : logy
```

Comment:
I'm not sure if these should be added, even if it makes the rule match finite differencing. NaN is usually not very useful as it infects everything.  So eg the forward rule will always yield NaN for any of these conditions. See eg https://juliadiff.org/ChainRulesCore.jl/stable/maths/nondiff_points.html.

Reply:
This is a bit mysterious ti me. On the (seemingly equivalent) branches in the rrule for logpdf of NegativeBinomial we wanted NaN. Was this something specific to Distributions.jl?

Reply:
I'm happy to change this but I'm unsure about the discrepancy to Distributions.jl